widgetssdk/src/main/java/com/glia/widgets/view/unifiedui/exstensions/MergeExtensions.kt

```diff
@@ -0,0 +1,28 @@
+package com.glia.widgets.view.unifiedui.exstensions
+
+import kotlin.reflect.KClass
+import kotlin.reflect.KParameter
+import kotlin.reflect.KProperty1
+import kotlin.reflect.full.declaredMemberProperties
+import kotlin.reflect.full.primaryConstructor
+
+private fun <T> mergeProperties(property: KProperty1<out T, Any?>, current: T, other: T): Any? {
+    val leftValue = property.getter.call(current)
+    val rightValue = property.getter.call(other)
+    return rightValue?.let { leftValue?.merge(it) } ?: rightValue ?: leftValue
+}
+
+infix fun <T : Any?> T.merge(other: T): T {
+    val current = this ?: return other
+    val nameToProperty = current::class.declaredMemberProperties.associateBy { it.name }
+    val primaryConstructor = current::class.primaryConstructor!!
+    val args: Map<KParameter, Any?> = primaryConstructor.parameters.associateWith { parameter ->
+        val property = nameToProperty[parameter.name]!!
+        val type = property.returnType.classifier as KClass<*>
+        when {
+            type.isData -> mergeProperties(property, current, other)
+            else -> property.getter.run { call(other) ?: call(current) }
+        }
+    }
+    return primaryConstructor.callBy(args)
+}
```

Comment:
In my opinion it might make sense to add comments or javadoc to this function. I'm not sure I understand all the logic here completely.

Besides, it should be good to have unit tests for this function, to make sure it works as expected for various possible cases.